### PR TITLE
[WIP] V2.0 - Federated Plugins, Web support

### PR DESCRIPTION
**Pull request by hnvn**
Introduce new v2:
- [x] Refactor codes to [Federated Plugins Implementation](https://docs.flutter.dev/development/packages-and-plugins/developing-packages#federated-plugins)
- [x] Web support

### CODE REVIEW

**Comment by hnvn** [comment]
Proposal for Web implementation:
- [croppie-dart](https://gitlab.com/michel.werren/croppie-dart)
- import JS library [import_js_library](https://pub.dev/packages/import_js_library)

**Comment by hnvn** [comment]
It seems that the plugin cannot return cropped image result as a `File` object on Web:
https://github.com/hnvn/flutter_image_cropper/blob/a6708c4d7e75d9b34763bd1aeb3fb9fe989235cd/image_cropper_platform_interface/lib/src/platform_interface/image_cropper_platform.dart#L75
Need a new result model to support both of `File` and raw byte array data `Uint8List`

**Comment by hnvn** [comment]
Completed Web implementation

**Comment by mono0926** [comment]
@hnvn 

https://github.com/mono0926/flutter_image_cropper/commit/dfa9fee5531083952679aa8e037a857b6c3f2026 is required for this kind of `generated_plugin_registrant.dart` to be generated.


```dart

import 'package:image_cropper_for_web/image_cropper_for_web.dart';

import 'package:flutter_web_plugins/flutter_web_plugins.dart';

// ignore: public_member_api_docs
void registerPlugins(Registrar registrar) {
  ImageCropperPlugin.registerWith(registrar);
}
```

File: image_cropper_for_web/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  js: ^0.6.3
+
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  flutter_lints: ^1.0.0
+
+flutter:
```
Comment (mono0926) [inline image_cropper_for_web/pubspec.yaml:26]: @hnvn 

https://github.com/mono0926/flutter_image_cropper/commit/dfa9fee5531083952679aa8e037a857b6c3f2026 is required for this kind of `generated_plugin_registrant.dart` to be generated.


```dart

import 'package:image_cropper_for_web/image_cropper_for_web.dart';

import 'package:flutter_web_plugins/flutter_web_plugins.dart';

// ignore: public_member_api_docs
void registerPlugins(Registrar registrar) {
  ImageCropperPlugin.registerWith(registrar);
}
```

Comment (hnvn) [reply image_cropper_for_web/pubspec.yaml:26]: So, it should be generated automatically, right?

Comment (mono0926) [reply image_cropper_for_web/pubspec.yaml:26]: @hnvn 

> So, it should be generated automatically, right?

Yes👌

**Pull request merged**